gitbook-docs/market/cmd-reference/statement.md

```diff
@@ -1,75 +1,104 @@
+Circuits are stored as a *statement* structure on the Proof Market.
+Statement examples can be found in `./example/statements/` directory.
+Attach all additional metadata to circuits generated by the zkLLVM compiler
+before publishing them on the Proof Market.
 
-**Usage**
+### Usage
 
-```
-python3 prepare_statement.py -h 
-```
+To see all parameters that can be passed to the script, run it with the `-h` option:
 
-Output
+```console
+$ python3 prepare_statement.py -h 
 
-```
-usage: prepare_statement.py [-h] -c CIRCUIT -o output file -n NAME -t TYPE
+usage: prepare_statement.py [-h] -c CIRCUIT -o output_file_path -n NAME -t \
+       statement_type (--private | --public)
 
 options:
   -h, --help            show this help message and exit
   -c CIRCUIT, --circuit CIRCUIT
-                        zkllvm compiler output
-  -o output file, --output output file
-                        output file
-  -n NAME, --name NAME  name
-  -t TYPE, --type TYPE  statement type: placeholder-zkllvm or placeholder-vanilla
-
+                        Path to a zkLLVM circuit (*.ll)
+  -o output_file_path, --output output_file_path
+                        Output file
+  -n NAME, --name NAME  Statement name that will be shown on the Proof Market
+  -t statement_type, --type statement_type
+                        Statement type: placeholder-zkllvm or placeholder-vanilla
+  --private             Make a statement private: only accessible by ID
+  --public              Make a statement public: visible on the Proof Market
 ```
 
-Ex:
+Please note that `--private` and `--public` parameters are mutually exclusive:
+* With `--private`, the statement will be accessible only by its ID.
+  For experiments and development purposes, make your statements private.
+* With `--public`, the statement will be openly listed on the Proof Market.
 
-```
-python3 prepare_statement.py -c arithmetics_example.ll -n arithmetic -t placeholder-zkllvm -o arithmetic.json
+### Example
+
+```bash
+python3 prepare_statement.py -c arithmetics_example.ll -n arithmetic -t placeholder-zkllvm -o \
+arithmetic.json --private
 ```
 
-### Publish Statement
+## Publishing statement
 
-Statements/Circuits can be published to the proof market once modified into the required format. See [prepare statement above](statement.md#prepare-statement).
+Statements/circuits can be published to the Proof Market once modified into the required format.
+See the [preparing statement](statement.md#preparing-circuit-as-a-statement) section above.
 
-#### Usage
+### Usage
+
+To publish a statement, you'll need to pass its JSON statement description in a file:
+
+```console
+$ python3 statement_tools.py push -h
 
-```
 usage: statement_tools.py push [-h] --file FILE
 
 options:
   -h, --help   show this help message and exit
   --file FILE  file
-
 ```
 
-#### Ex:
+You will be returned an object containing a *_key* field — a unique descriptor of the statement.
```

Comment:
How exactly will the user receive this object? I don't see it in the console output.

Reply:
This is a line from our [README](https://github.com/nilfoundation/proof-market-toolchain#2-circuit-generationpublishing-for-circuit-developers). Shall I remove it from here?

Reply:
I see in your examples the following output:
> Statement from /opt/zkllvm-template/build/template.json was pushed with id 12345678
So the id is returned, only not in object. Shall I refrase it like this then?
```suggestion
If successful, you will be returned an id of the statement.
```

Reply:
Use the text from template's readme: https://github.com/NilFoundation/zkllvm-template#step-5-publish-the-circuit-statement

Reply:
Updated